src/main/java/com/ontotext/kafka/error/LogErrorHandler.java

```diff
@@ -1,12 +1,93 @@
 package com.ontotext.kafka.error;
 
-import java.util.Collection;
-
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
 import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Properties;
+
+import static com.ontotext.kafka.GraphDBSinkConfig.ERRORS_DEAD_LETTER_QUEUE_TOPIC_NAME;
+import static com.ontotext.kafka.util.PropertiesUtil.*;
 
 public class LogErrorHandler implements ErrorHandler {
 
+	private static final Logger LOGGER = LoggerFactory.getLogger(LogErrorHandler.class);
+	private static final FailedRecordProducer PRODUCER = new FailedRecordProducer(FailedRecordProducer.getProperties());
+
 	@Override
-	public void handleFailingRecord(SinkRecord record) {}
+	public void handleFailingRecord(SinkRecord record, Throwable ex) {
+		String msg = record.toString();
+		logWarning("Record failed: " + msg, ex);
+		PRODUCER.returnFailed(record);
+	}
+
+	public static void logError(String msg, Throwable ex) {
+		LOGGER.error(msg, ex);
+	}
+
+	public static void logError(String msg) {
+		LOGGER.error(msg);
+	}
+
+	public static void logWarning(String msg, Throwable ex) {
+		LOGGER.warn(msg, ex);
+	}
+
+	public static void logWarning(String msg) {
+		LOGGER.warn(msg);
+	}
+
+	public static void logInfo(String msg) {
+		LOGGER.info(msg);
+	}
+
+	private static class FailedRecordProducer {
+		private final Properties properties;
+
+		FailedRecordProducer(Properties properties) {
+			this.properties = properties;
+		}
+
+		void returnFailed(SinkRecord record) {
+			String recordKey = record.key() == null ? "null" : record.key().toString();
+			String recordValue = record.value() == null ? "null" : record.value().toString();
+			Producer<String, String> producer = new KafkaProducer<>(properties);
+
+			try {
+				ProducerRecord<String, String> pr = new ProducerRecord<>(ERRORS_DEAD_LETTER_QUEUE_TOPIC_NAME, recordKey, recordValue);
+				producer.send(pr, (metadata, exception) -> {
+					String failedRecordInfo = String.format("Record (key=%s value=%s)\n" +
```

Comment:
Create a new ProducerErrorHandler that actually sends back the record. The purpose of the log error handler is to simply publish it in the log. That is why we have the ErrorHandler API and the records processor can be injected with either, depending on the selected config property. + add tests for it.